x-pack/plugins/apm/server/routes/services/get_service_transaction_groups_alerts.ts

```diff
@@ -0,0 +1,143 @@
+  SERVICE_NAME,
+  TRANSACTION_NAME,
+  TRANSACTION_TYPE,
+} from '../../../common/es_fields/apm';
+import { LatencyAggregationType } from '../../../common/latency_aggregation_types';
+import { AggregationType } from '../../../common/rules/apm_rule_types';
+import { environmentQuery } from '../../../common/utils/environment_query';
+import { ApmAlertsClient } from '../../lib/helpers/get_apm_alerts_client';
+import { MAX_NUMBER_OF_TX_GROUPS } from './get_service_transaction_groups';
+
+interface ServiceTransactionGroupAlertsAggResponse {
+  buckets: Array<
+    AggregationsFilterAggregate & {
+      key: string;
+      alerts_count: AggregationsCardinalityAggregate;
+    }
+  >;
+}
+
+export type ServiceTransactionGroupAlertsResponse = Array<{
+  name: string;
+  alertsCount: number;
+}>;
+
+const RuleAggregationType = {
+  [LatencyAggregationType.avg]: AggregationType.Avg,
+  [LatencyAggregationType.p99]: AggregationType.P99,
+  [LatencyAggregationType.p95]: AggregationType.P95,
+} as const;
+
+export async function getServiceTranactionGroupsAlerts({
+  apmAlertsClient,
+  kuery,
+  transactionType,
+  serviceName,
+  latencyAggregationType,
+  start,
+  end,
+  environment,
+}: {
+  apmAlertsClient: ApmAlertsClient;
+  kuery?: string;
+  serviceName?: string;
+  transactionType?: string;
+  latencyAggregationType: LatencyAggregationType;
+  start: number;
+  end: number;
+  environment?: string;
+}): Promise<ServiceTransactionGroupAlertsResponse> {
+  const ALERT_RULE_PARAMETERS_AGGREGATION_TYPE = `${ALERT_RULE_PARAMETERS}.aggregationType`;
+
+  const params = {
+    size: 0,
+    query: {
+      bool: {
+        filter: [
+          ...termQuery(ALERT_RULE_PRODUCER, 'apm'),
+          ...termQuery(ALERT_STATUS, ALERT_STATUS_ACTIVE),
+          ...rangeQuery(start, end),
+          ...kqlQuery(kuery),
+          ...termQuery(SERVICE_NAME, serviceName),
+          ...termQuery(TRANSACTION_TYPE, transactionType),
+          ...environmentQuery(environment),
+        ],
+        must: [
+          {
+            bool: {
+              should: [
+                ...termQuery(
+                  ALERT_RULE_PARAMETERS_AGGREGATION_TYPE,
+                  RuleAggregationType[latencyAggregationType]
+                ),
+                {
+                  bool: {
+                    must_not: [
+                      {
+                        exists: {
+                          field: ALERT_RULE_PARAMETERS_AGGREGATION_TYPE,
+                        },
+                      },
+                    ],
+                  },
+                },
+              ],
+            },
+          },
+        ],
+      },
+    },
+    aggs: {
+      transaction_groups: {
+        terms: {
+          field: TRANSACTION_NAME,
+          size: MAX_NUMBER_OF_TX_GROUPS,
+          order: { _count: 'desc' },
+        },
+        aggs: {
+          alerts_count: {
+            cardinality: {
+              field: ALERT_UUID,
+            },
+          },
+        },
```

Comment:
Do we need this sub-agg? Isn't the doc_count returned from the terms agg sufficient?